src/qnngds/devices/nanowire.py

```diff
@@ -24,19 +26,20 @@ def spot(
     Returns:
         Device: A device containing 2 optimal steps joined at their channel_w end.
     """
-
-    NANOWIRE = Device()
+    NANOWIRE = QnnDevice()
     wire = pg.optimal_step(channel_w, source_w, symmetric=True, num_pts=num_pts)
     source = NANOWIRE << wire
     gnd = NANOWIRE << wire
     source.connect(source.ports[1], gnd.ports[1])
 
-    NANOWIRE = pg.union(NANOWIRE, layer=layer)
+    #NANOWIRE = pg.union(NANOWIRE, layer=layer)
```

Comment:
you'd probably need to implement a union method for QnnDevice. It might make sense to have QnnDevice be a subclass of Device so you don't have to explicitly write out a union method (like you did with simplify), it would be inherited from the parent class.

Reply:
wait sorry I'm confused. Ignore this for now

Reply:
Really this is related to the broader question of modularity -- if we solved that some other way, we could revert to vanilla Devices

Reply:
I see, yeah I think I was mixing up QnnDevice and Design, hence why this comment is in kind of a random place hahaha. That's a good point also about phidl methods that return a Device. Guess we'd just have to override/redefine those methods anyway then.